dev_apps/in_app_pay_test/js/app.js

```diff
@@ -0,0 +1,166 @@
+    });
+  }
+
+  function fetch(options) {
+    return new Promise((resolve, reject) => {
+      var req = new XMLHttpRequest({mozSystem: true});
+      req.open(options.method, SERVER + options.url, true);
+      req.setRequestHeader('Content-Type', 'application/json');
+      req.responseType = 'json';
+      req.timeout = TIMEOUT;
+      req.withCredentials = true;
+      req.onload = () => {
+        DEBUG && console.log('Response to request', options.url, req.response);
+        if (req.status !== 201 && req.status !== 200 && req.status !== 204 &&
+            req.status !== 207 && req.status !== 302) {
+          reject(req.response);
+          return;
+        }
+        resolve(req.response);
+      };
+
+      req.onerror = req.ontimeout = (event) => {
+        DEBUG && console.log('Response to request', options.url, 'was an error');
+        reject(event.target.status);
+      };
+
+      var body;
+      if (options.body) {
+        body = JSON.stringify(options.body);
+      }
+
+      DEBUG && console.log('Request to', options.url, 'with body', body);
+      req.send(body);
+    });
+  }
+
+  function getJWT(server) {
+    return fetch({
+      method: 'GET',
+      url: '/jwt-source?server=' + server + '&_=' + Date.now()
+    });
+  }
+
+  var App = {
+    run: function() {
+      ['jwt-input',
+       'server',
+       'error',
+       'jwt',
+       'refresh',
+       'log',
+       'log-content',
+       'jwt-panel',
+       'jwt-content',
+       'pay',
+       'toggle'].forEach(id => {
+        this[toCamelCase(id)] = document.getElementById(id);
+      });
+
+      refresh.addEventListener('click', App.refresh);
+      pay.addEventListener('click', App.pay);
+      toggle.addEventListener('click', () => {
+        if (jwtPanel.classList.contains('hidden')) {
+          App.showJWT();
+        } else {
+          App.showLog();
+        }
+      });
+      server.addEventListener('change', App.refresh);
+      App.refresh();
+    },
+
+    pay: function() {
+      App.clearLog();
+      App.showLog();
+      App.writeLog('Signing JWT...');
+      fetch({
+        method: 'POST',
+        url: '/pay',
+        body: {
+          jwt: jwtContent.textContent,
+          server: server.value
+        }
+      }).then((response) => {
+        App.writeLog('[Transaction ID] -> ' + response.transID);
+        App.writeLog('Calling navigator.mozPay...');
+        return new Promise((resolve, reject) => {
+          var req = navigator.mozPay([response.jwt]);
+          req.onsuccess = () => {
+            resolve(response.transID);
+          };
+          req.onerror = () => {
+            reject('Payment canceled ' + req.error.name);
+          };
+        });
+      }).then((transactionID) => {
+        App.writeLog('Payment done!');
+        App.verifyTransaction(transactionID, 0);
+      }).catch((error) => {
+        App.writeLog('Oh crap ' + error);
+      });
+    },
+
+    verifyTransaction(transactionID, retries) {
```

Comment:
this syntax looks weird to me. Did you mean ``verifyTransaction: function() ...`` ?